samlcontent should be decrypted SAML response (sign then encrypt)

mastermatt opened the pull request:
This is a copy of the changes in https://github.com/tngan/samlify/pull/575, with the addition of test updates.

mastermatt committed:
samlcontent should be decrypted SAML response (sign then encrypt)

mastermatt commented:
@tngan @ahacker1-securesaml I've updated tests for this small change.
I would really appreciate it if we could get this out as a patch.

ahacker1-securesaml reviewed (approved):
looks good to me. as a note, users should not use samlContent for any securit purposes or use it for authentication details. it should be used for purely debugging purposes

ryanhos commented:
@tngan I have some discretionary sponsorship funds available for a one-time contribution. Any chance you have time to get eyes on this soon?

mastermatt committed:
Merge branch 'master' into samlcontent-sign-then-encrypt

# Conflicts:
#	test/flow.ts

mastermatt committed:
Call checkSignature is passed in arg

Passing a value that was parsed and re-stringified can mess up whitespace encodings.

mastermatt committed:
Merge branch 'master' into samlcontent-sign-then-encrypt

mastermatt committed:
Merge branch 'master' into samlcontent-sign-then-encrypt

# Conflicts:
#	src/flow.ts
#	src/libsaml.ts

mastermatt committed:
align variable pattern

mastermatt commented:
@tngan I've brought this PR up to date with the recent updates and confirmed the tests are still good. Could we get this merged in please? 
It ensures `samlContent` on a FlowResult is consistent after decryption regardless of signing order.
